### PR TITLE
[BACKLOG-5838] Prompt API - Implemented showProgressIndicator

### PR DESCRIPTION
**Pull request by annapopova1**
Sample framework: https://github.com/pentaho/pentaho-platform-ee/pull/860
@diogofscmariano @plagoa @krivera-pentaho @pamval  @afrjorge  please review

### CODE REVIEW

File: package-res/resources/web/prompting/api/UiAPI.js [hunk 1]
```diff
@@ -15,8 +15,25 @@
...
+ * Provides general functions for working with UI of prompting.
+ *
+ * @name UiAPI
+ * @class
+ */
 define([], function() {
   return function(api) {
-
+    /**
+     * Makes visible the progress indicator by calling function Dashboard#showProgressIndicator.
```
Comment (krivera-pentaho) [inline package-res/resources/web/prompting/api/UiAPI.js]: I would change this line to "Makes the progress indicator visible by calling Dashboard#showProgressIndicator"

File: package-res/resources/web/prompting/api/UiAPI.js [hunk 2]
```diff
@@ -15,8 +15,25 @@
  *
  */
 
+/**
+ * The Prompting UI API Class
+ * Provides general functions for working with UI of prompting.
```
Comment (krivera-pentaho) [inline package-res/resources/web/prompting/api/UiAPI.js]: Change to "Provides general functions for working with the UI of prompting."

**Comment by annapopova1** [comment]
@krivera-pentaho 
Virginia suggested another sentences for this API:
for class:
"This is the prompting user interface API class. Contains all general functions for working with the prompting user interface."

for method:
"Shows the progress indicator by calling the function Dashboard#showProgressIndicator."

Do you agree?

File: package-res/resources/web/prompting/api/UiAPI.js [hunk 3]
```diff
@@ -15,8 +15,25 @@
...
+    /**
+     * Makes visible the progress indicator by calling function Dashboard#showProgressIndicator.
+     *
+     * @name UiAPI#showProgressIndicator
+     * @method
+     * @example
+     *     api.ui.showProgressIndicator();
+     */
+    this.showProgressIndicator = function() {
+      api.operation._getPromptPanel().dashboard.showProgressIndicator();
```
Comment (pamval) [inline package-res/resources/web/prompting/api/UiAPI.js]: I'd probably avoid dashboard references to creep up to this level of the API.
This code assumes that a prompt panel will always contain a dashboard object that implements a showProgressIndicator method.

If we are targeting being able to change the prompt renderer from CDF to something else, that would mean that we'd have to change this function to achieve that, which feels wrong.

The easy fix would be for promptPanel to encapsulate how it shows/hides the progress indicator, therefore making the dashboard reference internal to it.

Comment (krivera-pentaho) [inline package-res/resources/web/prompting/api/UiAPI.js]: I completely agree with this.

**Comment by krivera-pentaho** [comment]
Yes Anna, I agree with the suggested changes.

**Comment by annapopova1** [comment]
@pamval @krivera-pentaho could you take a look?
- changed docs as Virginia proposed;
- added showProgressIndicator function to PromptPanel to encapsulate dashboard's logic.

**Comment by annapopova1** [comment]
changed docs as Virginia proposed one more time

**Comment by diogofscmariano** [comment]
@annapopova1 can we have the documentation extended with what we described in the meeting? Something like when this is called, what this does

**Comment by diogofscmariano** [comment]
@annapopova1 do we have these functions used in the prompt panel? does it makes sense to make sure within the prompt panel only these are being used and not others?

**Comment by annapopova1** [comment]
@diogofscmariano extended documentation.
This functions are not used in PromptPanel. I put it to PromptPanel for encapsulate operations with dashboard object (depends of CDF implementation). Pedro wrote the comment about it above.

**Comment by annapopova1** [comment]
@diogofscmariano  Implemented OperationAPI#refreshPrompt

**Comment by diogofscmariano** [comment]
@annapopova1 can't merge it. please fix the conflicts

**Comment by annapopova1** [comment]
@diogofscmariano fixed conflicts

**Pull request merged**